src/components/Company/Offers/Manage/CompanyOffersTitle.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { Chip } from "@material-ui/core";
+
+export const statusChips = {
+    hidden: <Chip label="Hidden" size="small" style={{ backgroundColor: "#90A4AE" }} />,
+    blocked: <Chip label="Blocked" size="small" style={{ backgroundColor: "#DC4338" }} />,
+    archived: <Chip label="Archived" size="small" style={{ backgroundColor: "#56A8D6" }} />,
+};
+
+const OfferTitle = ({ title, offersVisibility, setOfferVisibility, offerId }) => {
+    const [chips, setChips] = useState([]);
+    const isHidden = offersVisibility[offerId]?.isHidden;
+    const isBlocked = offersVisibility[offerId]?.isDisabled;
+    const isArchived = offersVisibility[offerId]?.isArchived;
```

Comment:
This code screams "REACT CONTEXT" to me. I know we talked about this in a meeting where we thought it un-wise to use it now because it would mean replacing the entire table implementation we use, but if you do come up with a solution please do let us know.